ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java

```diff
@@ -0,0 +1,782 @@
+})
+public class BaseListFragmentTest {
+
+    private TestBaseListFragment testBaseListFragment;
+    private Boolean isMethodSuccessfullyRun;
+
+    @Mock
+    private Looper looper;
+    @Mock
+    private View mockMRoot, mockView, mockCreateView;
+    @Mock
+    private RecyclerView mockMRecyclerView;
+    @Mock
+    private LayoutInflater mockInflater;
+    @Mock
+    private ViewGroup mockContainer;
+    @Mock
+    private Bundle savedInstanceState;
+    @Mock
+    private LinearLayoutManager mockLinearLayoutManager;
+    @Mock
+    private DefaultStateViewHelper mockMDefaultStateViewHelper;
+    @Mock
+    private CustomStateViewHelper mockCustomStateViewHelper;
+    @Mock
+    private DefaultStateViewHelper mockDefaultStateViewHelper;
+    @Mock
+    private OnListPageStateChangeListener mockListPageChangeStateListener;
+    @Mock
+    private Context mockContext;
+    @Mock
+    private EndlessRecyclerViewScrollAdapter mockAdapter;
+    @Mock
+    private Handler handler;
+    @Captor
+    private ArgumentCaptor<Runnable> runnableCaptor;
+    @Mock
+    private RecyclerView.OnScrollListener onScrollListener;
+    @Mock
+    private EndlessRecyclerViewScrollAdapter.OnLoadMoreListener loadMoreListener;
+    @Mock
+    private EndlessRecyclerViewScrollListener mloadMoreListener;
+
+    @Before
+    public void setUp() {
+        isMethodSuccessfullyRun = Boolean.FALSE;
+        MockitoAnnotations.initMocks(this);
+        testBaseListFragment = new TestBaseListFragment();
+        Whitebox.setInternalState(testBaseListFragment, "mCustomStateViewHelper", mockCustomStateViewHelper);
+        Whitebox.setInternalState(testBaseListFragment, "mDefaultStateViewHelper", mockDefaultStateViewHelper);
+        Whitebox.setInternalState(testBaseListFragment, "mListPageChangeStateListener", mockListPageChangeStateListener);
+        PowerMockito.mockStatic(Looper.class);
+        when(Looper.getMainLooper()).thenReturn(looper);
+    }
+
+    @Test
+    public void onCreateViewTest() {
+        //Arrange
+        BaseListFragment mockBaseListFragment = Mockito.mock(BaseListFragment.class);
+        Mockito.doCallRealMethod().when(mockBaseListFragment).onCreateView(mockInflater, mockContainer, savedInstanceState);
+        Mockito.doReturn(mockCreateView).when(mockInflater).inflate(R.layout.ko__fragment_list, mockContainer, false);
+        Mockito.doReturn(mockMRecyclerView).when(mockCreateView).findViewById(R.id.ko__list);
+        PowerMockito.suppress(methods(BaseListFragment.class, "setupView"));
+        //Act
+        View view  = mockBaseListFragment.onCreateView(mockInflater, mockContainer, savedInstanceState);
+        //Assert
+        Assert.assertNotNull(view);
+    }
+
+    class TestBaseListFragment extends BaseListFragment {
+
+        public TestBaseListFragment(){
+            super.mRoot = mockMRoot;
+            super.mRecyclerView = mockMRecyclerView;
+        }
+
+        @Override
+        public void setupView(View view) {
```

Comment:
✅ It is nice to see the usage of @Override taking advantage of compiler checking to make sure that the code is actually overriding a method.

**No change suggested, looks good!**
 

<a id="e0397f70-50d9-11e8-89e2-ed64d7ac7be5" href="none" codereview-uuid-reserved-tag />